internal/pkg/gateway/commit/blocknotifier.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package commit
+
+import (
+	"sync"
+
+	"github.com/hyperledger/fabric/core/ledger"
+)
+
+type blockListener interface {
+	receiveBlock(*ledger.CommitNotification)
+	close()
+}
+
+type blockNotifier struct {
+	commitChannel <-chan *ledger.CommitNotification
```

Comment:
I feel that too many `notifiers` are making it hard to read code. Can we club the functionality in this into the main `Notifier`. That will make the main `Notifier` act as a central piece that glues together the lower level events received from ledger and the other two notifiers handle the event specific delivery.

Reply:
I was thinking in the lines of launching a gorouting for each channel directly into the `Notifier` and the notifier just keeps track of the two event specific notifiers. In the current structure of the code it sounded like all three notifiers are kind of siblings when they are actually not. The other approach could be to have the both event specific notifiers only inside this `blocknotifier` - so this becomes full controller of channel specific notifier and hierarchy is clearer.

Reply:
Feel free to make that change